cli/mender-artifact/partition.go

```diff
@@ -53,37 +62,213 @@ type partition struct {
 	offset string
 	size   string
 	path   string
-	name   string
 }
 
-type vFile int
+type ModImageBase struct {
+	path string
+}
+
+type ModImageArtifact struct {
+	ModImageBase
+	*unpackedArtifact
+	comp artifact.Compressor
+	key  []byte
+}
+
+type ModImageSdimg struct {
+	ModImageBase
+	candidates []partition
+}
+
+type ModImageRaw struct {
+	ModImageBase
+}
+
+type vImage int
+
+type vImageAndFile struct {
+	image VPImage
+	file  VPFile
+}
 
 // Open is a utility function that parses an input image and path
 // and returns a V(irtual)P(artition)File.
-func (v vFile) Open(comp artifact.Compressor, imgpath string) (VPFile, error) {
-	imgname, fpath, err := parseImgPath(imgpath)
+func (v vImage) Open(comp artifact.Compressor, key []byte, imgname string) (VPImage, error) {
+	// first we need to check  if we are having artifact or image file
+	art, err := os.Open(imgname)
+	if err != nil {
+		return nil, errors.Wrap(err, "can not open artifact")
+	}
+	defer art.Close()
+
+	aReader := areader.NewReader(art)
+	err = aReader.ReadArtifact()
+	if err == nil {
+		// we have VALID artifact,
```

Comment:
I don't understand what you mean. If `ReadArtifact` rejects the artifact, then it is definitely not a valid artifact, but also not definitely an sdimg, it could also be a raw filesystem. I suppose I could name `processSdimg` differently, but it does use partition parsing under the hood, even for raw filesystem images, so I reckon the naming is appropriate.